sources/services/googleapi/GoogleApiService.ts

```diff
@@ -0,0 +1,213 @@
+import { Credentials, OAuth2Client } from "google-auth-library";
+import { calendar_v3, google, oauth2_v2 } from "googleapis";
+
+import { EventAttendeeRole } from "../../core/enums/EventAttendeeRole";
+import { EventAttendeeStatus } from "../../core/enums/EventAttendeeStatus";
+import { EventType } from "../../core/enums/EventType";
+import { EventVisibility } from "../../core/enums/EventVisibility";
+
+import { Calendar } from "../../models/entities/Calendar";
+import { CalendarMember } from "../../models/entities/CalendarMember";
+import { Event } from "../../models/entities/Event";
+import { User } from "../../models/entities/User";
+
+import { CalendarMemberRole } from "../../core/enums/CalendarMemberRole";
+
+import { CalendarRepository } from "../../models/CalendarRepository";
+import { EventRepository } from "../../models/EventRepository";
+import { UserRepository } from "../../models/UserRepository";
+
+type ServiceCredentials = {
+    readonly clientId: string;
+    readonly clientSecret: string;
+    readonly redirectUri: string;
+    readonly scopes: string;
+};
+
+export class GoogleApiService {
+    private readonly oauth: OAuth2Client;
+
+    constructor(
+        private readonly eventRepository: EventRepository,
+        private readonly userRepository: UserRepository,
+        private readonly calendarRepository: CalendarRepository,
+        private readonly credentials: ServiceCredentials,
+    ) {
+        this.eventRepository = eventRepository;
+        this.userRepository = userRepository;
+        this.calendarRepository = calendarRepository;
+        this.oauth = new google.auth.OAuth2(
+            credentials.clientId,
+            credentials.clientSecret,
+            credentials.redirectUri,
+        );
+    }
+
+    public generateAuthUrl(): string {
+        return this.oauth.generateAuthUrl({
+            access_type: "offline", // "offline" to obtain a refresh_token
+            scope: this.credentials.scopes,
+        });
+    }
+
+    public async getCredentialsFromCode(code: string): Promise<Credentials> {
+        const response = await this.oauth.getToken(code);
+        return response.tokens;
+    }
+
+    public async getUserInfo(accessToken: string, refreshToken: string): Promise<oauth2_v2.Schema$Userinfoplus> {
+        const client = this.createOauthApiClient(accessToken, refreshToken);
+
+        const res = await client.userinfo.get();
+        return res.data;
+    }
+
+    public async fetchGoogleCalendars(calendarMembers: CalendarMember[], user: User): Promise<Calendar[]> {
```

Comment:
Si t'as moyen de faire en sorte qu'elle appelle une fonction dans un autre fichier (à coté dans le dossier) ce serait fou !
Pour alleger cette belle classe

Reply:
Sorry bro i don't speak french

Reply:
I'm not you bro, bro